apm-agent-core/src/test/java/co/elastic/apm/agent/configuration/CoreConfigurationTest.java

```diff
@@ -0,0 +1,86 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.configuration;
+
+import org.junit.jupiter.api.Test;
+import org.stagemonitor.configuration.ConfigurationRegistry;
+import org.stagemonitor.configuration.source.SimpleSource;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class CoreConfigurationTest {
+
+    @Test
+    void testWithoutDisabledAndEnabledInstrumentations() {
+        CoreConfiguration config = getCoreConfiguration("", "");
+        assertThat(config.isInstrumentationEnabled("foo")).isTrue();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("foo"))).isTrue();
+        assertThat(config.isInstrumentationEnabled(Arrays.asList("foo", "bar"))).isTrue();
+    }
+
+    @Test
+    void testWithDisabledInstrumentations() {
+        CoreConfiguration config = getCoreConfiguration("", "foo");
+        assertThat(config.isInstrumentationEnabled("foo")).isFalse();
+        assertThat(config.isInstrumentationEnabled("bar")).isTrue();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("foo"))).isFalse();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("bar"))).isTrue();
+        assertThat(config.isInstrumentationEnabled(Arrays.asList("foo", "bar"))).isFalse();
+    }
+
+    @Test
+    void testWithEnabledInstrumentations() {
+        CoreConfiguration config = getCoreConfiguration("foo", "");
+        assertThat(config.isInstrumentationEnabled("foo")).isTrue();
+        assertThat(config.isInstrumentationEnabled("bar")).isFalse();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("foo"))).isTrue();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("bar"))).isFalse();
+        assertThat(config.isInstrumentationEnabled(Arrays.asList("foo", "bar"))).isTrue();
+    }
+
+    @Test
+    void testWithDisabledAndEnabledInstrumentations() {
+        CoreConfiguration config = getCoreConfiguration("foo", "foo");
+        assertThat(config.isInstrumentationEnabled("foo")).isFalse();
+        assertThat(config.isInstrumentationEnabled("bar")).isFalse();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("foo"))).isFalse();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("bar"))).isFalse();
+        assertThat(config.isInstrumentationEnabled(Arrays.asList("foo", "bar"))).isFalse();
+    }
+
+    @Test
+    void testLegacyDefaultDisabledInstrumentation() {
+        CoreConfiguration config = getCoreConfiguration("", "incubating");
+        assertThat(config.isInstrumentationEnabled("experimental")).isFalse();
+        assertThat(config.isInstrumentationEnabled(Collections.singletonList("experimental"))).isFalse();
+    }
+
```

Comment:
Adding a test case for instrumentation that is experimental:
- adding it to `enabled_instrumentations` should not enable it by default
- adding it to `enabled_instrumentations` AND `enable_experimental_instrumentations` should selectively enable it, but not any of the other instrumentations (both experimental and non-experimental).

Reply:
Done!